### PR TITLE
Slackclient app

### PR DESCRIPTION
**Pull request by NGTmeaty**
This should be completing #11. It should move from webhooks to slackclient.

### CODE REVIEW

File: tor_core/helpers.py [hunk 1]
```diff
@@ -79,23 +79,23 @@ def clean_list(items):
     return cleaned
 
 
-def send_to_slack(message, config):
+def send_to_slack(message, config,  channel="#general"):
```
Comment (perryprog) [inline tor_core/helpers.py]: Extra whitespace here

**Review by perryprog — CHANGES_REQUESTED** [review]
Can you change the first two commit messages so they are different, or squash them?

File: tor_core/helpers.py (hunk 1 above)
Comment (perryprog) [reply tor_core/helpers.py]: Oh, there's also double quotes instead of single quotes here.

File: tor_core/helpers.py [hunk 2]
```diff
@@ -79,23 +79,23 @@ def clean_list(items):
...
-            'username': 'Kierra',
-            'icon_emoji': ':snoo:',
-            'text': message
-        }
-        requests.post(config.slack_api_url, json=payload)
+    if config.slack_api_key:
+        slackc = SlackClient(config.slack_api_key)
+
+        slackc.api_call(
+            "chat.postMessage",
```
Comment (perryprog) [inline tor_core/helpers.py]: Double quotes instead of single quotes here again

**Review by perryprog — CHANGES_REQUESTED** [review]

File: tor_core/helpers.py [hunk 3]
```diff
@@ -79,23 +79,23 @@ def clean_list(items):
...
 
-def send_to_slack(message, config):
+def send_to_slack(message, config,  channel="#general"):
     """
-    Sends a message to the ToR #general slack channel.
+    Sends a message to the ToR slack.
 
-    :param message: String; the message that is to be encoded.
+    :param message: String; the message that is to be encoded
+    :param channel: channel option, defaults to general
```
Comment (itsthejoker) [inline tor_core/helpers.py]: Let's add in that this should be a string as well -- we're not going full-on type hinting, but I like to have it in the docstrings.

**Comment by itsthejoker** [comment]
Overall it looks good -- let's change the double quotes to single quotes and we'll test. I don't really care about the doubled up commit messages; just something to watch out for in the future.

**Comment by NGTmeaty** [comment]
Changes should be made.

File: tor_core/helpers.py [hunk 4]
```diff
@@ -79,23 +79,23 @@ def clean_list(items):
     return cleaned
 
 
-def send_to_slack(message, config):
+def send_to_slack(message, config,  channel='#general'):
```
Comment (perryprog) [inline tor_core/helpers.py]: There is still whitespace here ):

**Review by perryprog — CHANGES_REQUESTED** [review]

**Review by perryprog — APPROVED** [review]
LGTM

**Review by itsthejoker — APPROVED** [review]

File: tor_core/helpers.py [hunk 5]
```diff
@@ -79,23 +80,23 @@ def clean_list(items):
     return cleaned
 
 
-def send_to_slack(message, config):
+def send_to_slack(message, config, channel='#general'):
```
Comment (TheLonelyGhost) [inline tor_core/helpers.py:83]: Just came in to comment, this (and below code) is a beautiful way of retaining backwards compatibility despite a lot of added functionality. Very nice job!

**Pull request merged**